### PR TITLE
Create daily itineraries

### PR DESCRIPTION
**Pull request by hoadam**
## Type of change
- [X] New feature
- [x] Bug Fix
- [x] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s):
*Brief summary of the change(s) included in this pull request*
Create/Delete daily_itineraries when user creates/edits dates of the trip

This description closes #

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [x] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*

## Checklist:
- [ ] My code has no unused/commented out code
- [ ] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

### CODE REVIEW

File: app/models/trip.rb [hunk 1]
```diff
@@ -11,9 +11,59 @@ class Trip < ApplicationRecord
    validates :total_budget, presence: true
    validates :user_id, presence: true
 
+   after_create_commit :create_daily_itineraries
+   after_update_commit :update_daily_itineraries, if: :dates_changed?
    enum status: [:in_progress, :completed]
```
Comment (yainporter) [inline app/models/trip.rb:16]: Nice callback!

**Review by yainporter — APPROVED** [review]
Nice job Hoa, thanks for your work on bug catch!!

**Review by Selena730 — APPROVED** [review]
Nice work!
Great testing

**Review by IgorrMagalhaess — APPROVED** [review]
Amazing work

**Pull request merged**